docs/source/tutorials/improving_ml_performance.ipynb

```diff
@@ -786,84 +786,85 @@
+    "    \"\"\"\n",
+    "    # Extract indices for each type of issue\n",
+    "    label_issues_idx = label_issues.index.tolist()\n",
+    "    near_duplicates_idx = near_duplicates.index.tolist()\n",
+    "    outliers_idx = outliers.index.tolist()\n",
+    "\n",
+    "    # Calculate the number of each type of data point to drop except near duplicates, which requires separate logic\n",
+    "    num_label_issues_to_drop = int(len(label_issues_idx) * drop_label_issue)\n",
+    "    num_outliers_to_drop = int(len(outliers_idx) * drop_outlier)\n",
+    "\n",
+    "    # Calculate number of near duplicates to drop\n",
+    "    # Assuming the 'near_duplicate_sets' are lists of indices (integers) of near duplicates\n",
+    "    clusters = []\n",
+    "    for i in near_duplicates_idx:\n",
+    "        # Create a set for each cluster, add the current index to its near duplicate set\n",
+    "        cluster = set(near_duplicates.at[i, 'near_duplicate_sets'])\n",
+    "        cluster.add(i)\n",
+    "        clusters.append(cluster)\n",
+    "\n",
+    "    # Deduplicate clusters by converting the list of sets to a set of frozensets\n",
+    "    unique_clusters = set(frozenset(cluster) for cluster in clusters)\n",
+    "\n",
+    "    # If you need the unique clusters back in list of lists format:\n",
+    "    unique_clusters_list = [list(cluster) for cluster in unique_clusters]\n",
+    "\n",
+    "    near_duplicates_idx_to_drop = []\n",
+    "\n",
+    "    for cluster in unique_clusters_list:\n",
+    "        # Calculate the number of rows to drop, ensuring at least one datapoint remains\n",
+    "        n_drop = max(math.ceil(len(cluster) * drop_near_duplicate), 1)  # Drop at least k% or 1 row\n",
+    "        if len(cluster) > n_drop:  # Ensure we keep at least one datapoint\n",
+    "            # Randomly select datapoints to drop\n",
+    "            drops = random.sample(cluster, n_drop)\n",
+    "        else:\n",
+    "            # If the cluster is too small, adjust the number to keep at least one datapoint\n",
+    "            drops = random.sample(cluster, len(cluster) - 1)  # Keep at least one\n",
+    "        near_duplicates_idx_to_drop.extend(drops)\n",
+    "\n",
+    "    # Determine the specific indices to drop\n",
+    "    label_issues_idx_to_drop = label_issues_idx[:num_label_issues_to_drop]\n",
+    "    outliers_idx_to_drop = outliers_idx[:num_outliers_to_drop]\n",
+    "\n",
+    "    # Combine the indices to drop\n",
+    "    idx_to_drop = list(set(label_issues_idx_to_drop + near_duplicates_idx_to_drop + outliers_idx_to_drop))\n",
+    "\n",
+    "    # Drop the rows from the training data\n",
+    "    train_features_cleaned = train_features.drop(idx_to_drop).reset_index(drop=True)\n",
+    "    train_labels_cleaned = train_labels.drop(idx_to_drop).reset_index(drop=True)\n",
+    "\n",
+    "    return train_features_cleaned, train_labels_cleaned\n",
+    "</details>\n",
+    "```"
```

Comment:
```suggestion
"```\n",
"\n",
"</details>\n",

```

Reply:
Isn't what you have here wrong?  Unclear to me how you were able to get the docs fixed with what you had?

What you should've done is referenced a similar tutorial where this is done, eg:
https://raw.githubusercontent.com/cleanlab/cleanlab/master/docs/source/tutorials/datalab/audio.ipynb

and copied the formatting there exactly.

Reply:
<img width="423" alt="Screenshot 2024-07-17 at 10 36 18 PM" src="https://github.com/user-attachments/assets/2cb59da2-a912-445f-a7d6-bd9f6f746aa8">

Oh yeah I did reference how this is done in another [notebook](https://github.com/cleanlab/cleanlab/blob/master/docs/source/tutorials/datalab/image.ipynb), so that's strange because this is what I have in the latest version of the notebook I committed in the screenshot. And it is what I built the docs locally with

Reply:
And then under Files Changed, you can see the markdown removed in latest commit had another indentation level present that is not present anymore